src/fabric_cli/commands/config/fab_config_set.py

```diff
@@ -102,3 +90,20 @@ def _set_capacity(args: Namespace, value: str) -> None:
             ErrorMessages.Config.invalid_capacity(value),
             fab_constant.ERROR_INVALID_INPUT,
         )
+
+
+def _handle_fab_config_mode(previous_mode: str, current_mode: str) -> None:
+        from fabric_cli.core.fab_context import Context
+        # Clean up context files when changing mode
+        Context().cleanup_context_files(cleanup_all_stale=True, cleanup_current=True)
+        
+        if (current_mode == fab_constant.FAB_MODE_INTERACTIVE
+            and previous_mode == fab_constant.FAB_MODE_COMMANDLINE):
+            utils_ui.print("Switching to interactive mode...")
+            from fabric_cli.core.fab_interactive import start_interactive_mode
+            start_interactive_mode()
```

Comment:
no need to check if previous mode is command line. let's remove `and previous_mode == fab_constant.FAB_MODE_COMMANDLINE)` and make sure the start_interactive_mode handles the case where i already in the interactive mode (meaning the InteractiveCLI instance exists)